### PR TITLE
Feature: Adding MVDream model.

### PR DESCRIPTION
**Pull request by jiexiong2016**
Added example about the MVDream, a model to perform text to multi-view(4) images generation.

### CODE REVIEW

File: examples/mvdream/models/mvdream.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        self.sampler = DDIMSampler(self.model)
+
+        # output 4 view with fixed angles
+        self.nviews = 4
+        self.camera = get_camera(self.nviews).to(self.device_str)
+
+    def __call__(
+        self,
+        prompts: Union[str, List[str]],
+        negative_prompts: Union[str, List[str]] = [""],
```
Comment (spillai) [inline examples/mvdream/models/mvdream.py:33]: [nit] Defaults to `None` which is identical to the the diffusers library. See how we do this in [stable_diffusion.py](https://github.com/autonomi-ai/nos/blob/main/nos/models/stable_diffusion.py).

File: examples/mvdream/models/mvdream.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    def __call__(
+        self,
+        prompts: Union[str, List[str]],
+        negative_prompts: Union[str, List[str]] = [""],       
+        num_inference_steps: int = 50,
+        guidance_scale: float = 7.5,
+        ddim_eta:float = 0.0,
+        image_size: int = 256,
+        seed: int = -1,
+    ) -> List[Dict[str, Any]]:
```
Comment (spillai) [inline examples/mvdream/models/mvdream.py:39]: Is the output a List of dictionaries (one for each prompt), where `key` is the view name, and `value` is the image?

File: examples/mvdream/serve.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+images:
+  mvdream-gpu:
+    base: autonomi/nos:0.1.0rc3-gpu
```
Comment (spillai) [inline examples/mvdream/serve.yaml:3]: You can use `autonomi/nos:0.1.0-gpu` from now on.

File: examples/mvdream/serve.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    pip:
+      - transformers==4.27.1
+      - xformers>=0.0.22
+      - open-clip-torch==2.7.0
+      - omegaconf
+      - einops
+    env:
```
Comment (spillai) [inline examples/mvdream/serve.yaml:10]: Great for debugging, but let's remove this for demos etc.

File: examples/mvdream/serve.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    run:
+      - pip install --upgrade pip
+      - pip install --no-deps git+https://github.com/bytedance/MVDream.git
+
+models:
+  mv-dream:
+    model_cls: MVDreamModel
+    model_path: models/mvdream.py
+    default_method: __call__
+    runtime_env: mvdream-gpu
```
Comment (spillai) [inline examples/mvdream/serve.yaml:21]: 💯

**Review by spillai — APPROVED** [review]
💯 added some comments, but overall 🔥

File: examples/mvdream/models/mvdream.py (hunk 1 above)
Comment (jiexiong2016) [reply examples/mvdream/models/mvdream.py:33]: True, changed to `None` now.

File: examples/mvdream/models/mvdream.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        
+        if type(prompts)!=list:
```
[the remark this answers is not in the document]
Comment (jiexiong2016) [reply examples/mvdream/models/mvdream.py:41]: Got it, using `isinstance` now.

File: examples/mvdream/models/mvdream.py (hunk 2 above)
Comment (jiexiong2016) [reply examples/mvdream/models/mvdream.py:39]: Was a bug :man_facepalming:, changed to `List[Image.Image]` now.

File: examples/mvdream/models/mvdream.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            prompts = [prompts]
+
+        if type(negative_prompts)!=list:
+            negative_prompts = [negative_prompts]
+            
+        with torch.no_grad(), torch.autocast(device_type=self.device_str, dtype=self.torch_dtype):
+            torch.manual_seed(seed)
+            
+            c = self.model.get_learned_conditioning(prompts).to(self.device_str)
```
[the remark this answers is not in the document]
Comment (jiexiong2016) [reply examples/mvdream/models/mvdream.py:55]: Added some comments here.

File: examples/mvdream/models/mvdream.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            uc = self.model.get_learned_conditioning(negative_prompts).to(self.device_str)
+
+            c_ = {"context": c.repeat(self.nviews, 1, 1)}
```
[the remark this answers is not in the document]
Comment (jiexiong2016) [reply examples/mvdream/models/mvdream.py:59]: Added some comments here.

**Pull request merged**